src/well_known_srs.cpp

```diff
@@ -36,57 +36,41 @@ MAPNIK_DISABLE_WARNING_POP
 
 namespace mapnik {
 
-extern std::string const MAPNIK_LONGLAT_PROJ =
-    "+proj=longlat +ellps=WGS84 +datum=WGS84 +no_defs";
+extern std::string const MAPNIK_GEOGRAPHIC_PROJ =
+    "epsg:4326";  //wgs84
 
-extern std::string const MAPNIK_GMERC_PROJ =
-    "+proj=merc +a=6378137 +b=6378137 +lat_ts=0.0 +lon_0=0.0 +x_0=0.0"
-    " +y_0=0.0 +k=1.0 +units=m +nadgrids=@null +wktext +no_defs +over";
+extern std::string const MAPNIK_WEBMERCATOR_PROJ =
+    "epsg:3857"; // webmercator
 
 static const char * well_known_srs_strings[] = {
-    "mapnik-longlat",
-    "mapnik-gmerc",
+    MAPNIK_GEOGRAPHIC_PROJ.c_str(),
+    MAPNIK_WEBMERCATOR_PROJ.c_str(),
     ""
 };
 
 boost::optional<well_known_srs_e> is_well_known_srs(std::string const& srs)
 {
-    if (srs == "+init=epsg:4326" || srs == MAPNIK_LONGLAT_PROJ)
+    if (srs == MAPNIK_GEOGRAPHIC_PROJ)
     {
         return boost::optional<well_known_srs_e>(mapnik::WGS_84);
     }
-    else if (srs == "+init=epsg:3857" || srs == MAPNIK_GMERC_PROJ)
+    else if (srs == MAPNIK_WEBMERCATOR_PROJ)
     {
-        return boost::optional<well_known_srs_e>(mapnik::G_MERC);
+        return boost::optional<well_known_srs_e>(mapnik::WEB_MERC);
     }
     return boost::optional<well_known_srs_e>();
 }
 
 boost::optional<bool> is_known_geographic(std::string const& srs)
 {
     std::string trimmed = util::trim_copy(srs);
-    if (trimmed == "+init=epsg:3857")
-    {
-        return boost::optional<bool>(false);
-    }
-    else if (trimmed == "+init=epsg:4326")
+    if (trimmed == MAPNIK_GEOGRAPHIC_PROJ)
     {
         return boost::optional<bool>(true);
     }
-    else if (srs.find("+proj=") != std::string::npos)
+    else if (trimmed ==  MAPNIK_WEBMERCATOR_PROJ)
     {
-        if ((srs.find("+proj=longlat") != std::string::npos) ||
-                          (srs.find("+proj=latlong") != std::string::npos) ||
-                          (srs.find("+proj=lonlat") != std::string::npos) ||
-                          (srs.find("+proj=latlon") != std::string::npos)
-                         )
-        {
-            return boost::optional<bool>(true);
-        }
-        else
-        {
-            return boost::optional<bool>(false);
-        }
+        return boost::optional<bool>(true);
```

Comment:
@artemp 
This was previously false. Looks like a typo caused by swapping the order of checks (was webmercator first).

Reply:
@lightmare great catch as usual! I'll fix it and add unit test, thanks.

Reply:
Fixed in 3a302667a1caf6831b6dc96c14e0729debbc1286 
Test added in 2bb5bdb9fe9e4f38e52f664e6cfda4dae8d94551